File: Novigo/ReverseString.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
void reverse(char s[]){
    int left=0;
    int right=strlen(s)-1;
    int temp;
    while(left<right){
        temp=s[left];
        s[left]=s[right];
        s[right]=temp;
        left++;
        right--;
    }
}
void reverseWithoutTemp(char s[]){
    int left=0;
    int right=strlen(s)-1;
    while(left<right){
       s[left]^=s[right];
       s[right]^=s[left];
       s[left]^=s[right];
       left++;
       right--;

    }

}

int main(){
    char s[100];
    printf("enter the string\n");
    scanf("%s",s);
    reverse(s);
    printf("%s\n",s);
    reverseWithoutTemp(s);
    printf("%s\n",s);

}

File: Novigo/ReverseString.py
def reverse(s):
    chars=list(s)
    left,right=0,len(s)-1
    while (left<right):
        chars[left],chars[right]=chars[right],chars[left]
        left=left+1
        right=right-1
    return ''.join(chars)
input_str="vishal"
result=reverse(input_str)
print(result)
        
    
    